developer-docs-site/docs/move/move-on-aptos/cryptography.md

```diff
@@ -0,0 +1,186 @@
+For example, one can verify BLS signatures over BLS12-381 curves by calling the function above with the BLS12-381 marker types as its type arguments:
+
+```move
+use aptos_std::bls12381_algebra::{
+    G1, G2, Gt, FormatG1Compr, FormatG2Compr, HashG1XmdSha256SswuRo
+};
+
+// Aborts with code 1 if the BLS signature over the BLS12-381 curve fails to verify. 
+assert(
+    bls_verify_sig<G1, G2, Gt, FormatG1Compr, FormatG2Compr, HashG1XmdSha256SswuRo>(
+        dst, signature, message, public_key
+    ),
+    1
+);
+```
+
+For more use cases of the `crypto_algebra` module, check out some Move examples:
+
+1. [Verifying Groth16 zkSNARK proofs](#groth16-zksnark-verifier) over **any** curve 
+2. [Verifying randomness from the `drand` beacon](#verifying-randomness-from-the-drand-beacon)
+
+## Building powerful cryptographic applications in Move
+
+### Veiled coins
+
+The [`veiled_coin` example](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/veiled_coin/sources) demonstrates how to use [the Ristretto255 modules from above](#ristretto255-arithmetic-in-move) to add a reasonable layer of confidentiality to coin balances and transactions.
+
+Specifically, users can **veil** their balance, keeping it hidden from everyone, including validators.
+Furthermore, a user can send a **veiled transaction** that hides the transaction amount from everybody, including validators.
+An important caveat is that veiled transactions do NOT hide the identities of the sender or the recipient.
+
+**WARNING:** This module is educational. It is **NOT** production-ready. Using it could likely lead to loss of funds.
+
+### Groth16 zkSNARK verifier
+
+The [`groth16` example](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/groth16_example/sources) demonstrates how to verify Groth16 zkSNARK proofs[^groth16], which are the shortest and fastest-to-verify zero-knowledge proofs for arbitrary computations.
+Importantly, as explained [above](#generic-elliptic-curve-arithmetic-in-move), this implementation is *generic* over **any** curve, making it very easy for Move developers to use it with their favorite (supported) curves.
+
+**WARNING:** This code has not been audited. If using it in a production system, proceed at your own risk.
+
+### Verifying randomness from the `drand` beacon
+
+The [`drand` example](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/drand/sources) shows how to verify public randomness from the [drand](https://drand.love) randomness beacon.
+This randomness can be used in games or any other chance-based smart contract.
+We give a simple example of a lottery implemented on top of `drand` randomness in [`lottery.move`](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/drand/sources/lottery.move).
+
+**WARNING:** This code has not been audited. If using it in a production system, proceed at your own risk.
```

Comment:
I think we should think about this statement, it has been reviewed, what is an audit but a review? I think this should be a bit more explicit in that this has not been audited by a third-party auditing organization or something along those lines.

Reply:
Fixed.